core/server/src/main/java/alluxio/master/AbstractMaster.java

```diff
@@ -251,11 +253,19 @@ protected void waitForJournalFlush(long counter) {
       return;
     }
     Preconditions.checkNotNull(mAsyncJournalWriter, PreconditionMessage.ASYNC_JOURNAL_WRITER_NULL);
-    try {
-      mAsyncJournalWriter.flush(counter);
-    } catch (IOException e) {
-      throw new RuntimeException(e);
+
+    // TODO(gpang): add configurable parameter
+    RetryPolicy retry = new TimeoutRetry(300000, 1000);
+    while (retry.attemptRetry()) {
+      try {
+        mAsyncJournalWriter.flush(counter);
+        return;
+      } catch (IOException e) {
+        LOG.warn("Journal flush failed. retrying...", e);
+      }
     }
+    LOG.error("Journal flush failed after retries. Terminating process to prevent inconsistency.");
```

Comment:
`after retries` is odd. Should we count the retries and print how many retries there were?